pkg/controller/volume/persistentvolume/pv_controller.go

```diff
@@ -396,6 +426,19 @@ func (ctrl *PersistentVolumeController) syncUnboundClaim(claim *v1.PersistentVol
 		if err != nil {
 			return err
 		}
+		// Clean up the waiting for consumer condition as we already have VolumeName set
+		if c := findStatusCondition(claim.Status.Conditions, v1.PersistentVolumeClaimWaitingForConsumer); c != nil && c.Status == v1.ConditionTrue {
+			newClaim := claim.DeepCopy()
+			newClaim.Status.Conditions = []v1.PersistentVolumeClaimCondition{}
+			for _, c := range claim.Status.Conditions {
+				if c.Type == v1.PersistentVolumeClaimWaitingForConsumer {
+					continue
+				}
+				newClaim.Status.Conditions = append(newClaim.Status.Conditions, c)
+			}
+			_, err := ctrl.GetKubeClient().CoreV1().PersistentVolumeClaims(claim.GetNamespace()).Update(context.TODO(), newClaim, metav1.UpdateOptions{})
+			return err
+		}
```

Comment:
Put this into a separate function, `syncUnboundClaim` is long enough already.

Reply:
done